spring-cloud-starter-stream-processor-twitter-sentiment/src/test/java/org/springframework/cloud/stream/app/twitter/sentiment/processor/twitter/TwitterSentimentTensorflowProcessorIntegrationTests.java

```diff
@@ -0,0 +1,91 @@
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.cloud.stream.app.twitter.sentiment.processor.twitter;
+
+import static org.hamcrest.Matchers.equalTo;
+
+import org.junit.Assert;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.autoconfigure.SpringBootApplication;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.cloud.stream.app.twitter.sentiment.processor.TwitterSentimentProcessorConfiguration;
+import org.springframework.cloud.stream.messaging.Processor;
+import org.springframework.cloud.stream.test.binder.MessageCollector;
+import org.springframework.context.annotation.Import;
+import org.springframework.integration.support.MessageBuilder;
+import org.springframework.messaging.Message;
+import org.springframework.test.annotation.DirtiesContext;
+import org.springframework.test.context.junit4.SpringRunner;
+
+/**
+ * Integration Tests for TwitterSentimentTensorflowProcessor
+ *
+ * @author Christian Tzolov
+ */
+@SuppressWarnings("SpringJavaAutowiringInspection")
+@RunWith(SpringRunner.class)
+@SpringBootTest(
+		webEnvironment = SpringBootTest.WebEnvironment.NONE,
+		properties = {
+				"tensorflow.modelLocation=http://dl.bintray.com/big-data/generic/minimal_graph.proto",
+				"tensorflow.outputName=output/Softmax",
+				"twitter.vocabularyLocation=http://dl.bintray.com/big-data/generic/vocab.csv"
+		})
+@DirtiesContext
+public abstract class TwitterSentimentTensorflowProcessorIntegrationTests {
+
+	@Autowired
+	protected Processor channels;
+
+	@Autowired
+	protected MessageCollector messageCollector;
+
+	public static class OutputInPayloadTests extends TwitterSentimentTensorflowProcessorIntegrationTests {
+
+		@Test
+		public void testEvaluationPositive() {
+			testEvaluation("{\"text\": \"RT @PostGradProblem: In preparation for the NFL lockout ...\", \"id\":666, \"lang\":\"en\" }",
+					"{\"sentiment\":\"POSITIVE\",\"text\":\"RT @PostGradProblem: In preparation for the NFL lockout ...\",\"id\":666,\"lang\":\"en\"}");
+		}
+
+		@Test
+		public void testEvaluationNegative() {
+			testEvaluation("{\"text\": \"This is really bad\", \"id\":666, \"lang\":\"en\" }",
+					"{\"sentiment\":\"NEGATIVE\",\"text\":\"This is really bad\",\"id\":666,\"lang\":\"en\"}");
+		}
+
+
+		private void testEvaluation(String tweetJson, String resultJson) {
+
+			channels.input().send(MessageBuilder.withPayload(tweetJson).build());
+
+			Message<String> received = (Message<String>) messageCollector.forChannel(channels.output()).poll();
```

Comment:
This cast does not make sense since you check for the type in the `assert` below.
And going down we see `toString()` which says that we even don't need to check for `String` type at all.
Each `Object` has `toString()` and `String.toString()` does exactly what we need:
```
public String toString() {
      return this;
}
```